lib/modules/manager/mix/extract.ts

```diff
@@ -28,26 +31,19 @@ export async function extractPackageFile(
       } while (!contentArr[lineNumber].includes('end'));
       let depMatchGroups = depMatchRegExp.exec(depBuffer)?.groups;
       while (depMatchGroups) {
-        const { depName, datasource, currentValue, organization } =
-          depMatchGroups;
+        const { app, requirement, opts } = depMatchGroups;
+        const github = githubRegexp.exec(opts)?.groups?.value;
+        const organization = organizationRegexp.exec(opts)?.groups?.value;
+        const datasource = github ? githubDataSourceId : HexDatasource.id;
 
         const dep: PackageDependency = {
-          depName,
-          currentValue,
+          depName: app,
+          currentValue: requirement,
+          datasource,
+          packageName: organization ? `${app}:${organization}` : app,
```

Comment:
this should be the github repo instead when defined

Reply:
Hm, I might misunderstood your messages in [this thread](https://github.com/renovatebot/renovate/pull/19128#discussion_r1035280564) then, as I understood you wanted the `packageName` to be equal to `depName` if it's a package hosted on GitHub (that we're skipping anyway).

Why would `packageName` be the name of the GitHub repo? For a dependency declaration like `{:cowboy, github: "ninenines/cowboy"}` the dependency name _is_ `cowboy`. `ninenines/cowboy` is just a source where to fetch the dependency. You could treat it as a version requirement, since it essentially resolves to a git commit and is saved in the lock file, but I don't think it should be the `packageName`. I don't know Renovate's codebase that well so it would help me if you could explain what you request a bit more.

Reply:
no depName steps same, but for github deps set the package name to the repo name.
leave current value undefined.

that ways we can some time in future support update-lockfile range strategy. assuming mix uses default branch to find commit to update.

so we no longer need to skip those deps inside manager.